test/commands/positive.test.ts

```diff
@@ -82,7 +119,137 @@ describe('Positive', () => {
+
+      expect(mockValidateParameters).toHaveBeenCalledTimes(3);
+      expect(mockValidateParameters).toHaveBeenCalledWith('walkIntoMordor', {
+        guide: 'gollum',
+      });
+      expect(mockValidateParameters).toHaveBeenCalledWith('getHobbit', {
+        name: 'Frodo',
+      });
+      expect(mockValidateParameters).toHaveBeenCalledWith('getTomBombadil', {
+        times: 2,
+      });
+    });
+
+    describe('parameter validation fails', () => {
+      it('outputs a failure for that operation', async () => {
+        mockValidateParameters.mockImplementation(
+          (operationId) =>
+            new Promise((resolve) => {
+              if (operationId === 'walkIntoMordor')
+                throw new TypeMismatchError(
+                  ['parameters', 'guide', 'example'],
+                  'string',
+                  'number',
+                );
+
+              resolve();
+            }),
+        );
+
+        await expect(async () => {
+          await Positive.run(['http://urldoesnotmatter.com']);
+        }).rejects.toThrow('1 operation failed');
+
+        expect(result).toEqual([
+          'walkIntoMordor: Failed - Actual type did not match schema. Path: parameters -> guide -> example. Schema type: string. Actual type: number\n',
+          'getHobbit: Succeeded\n',
+          'getTomBombadil: Succeeded\n',
+        ]);
+      });
+    });
+
     describe('operation has parameter groups', () => {
+      beforeEach(() => {
+        mockGetParameters.mockImplementation(
+          () =>
+            new Promise((resolve) =>
+              resolve({
+                walkIntoMordor: [
+                  {
+                    door: 'front',
+                  },
+                  {
+                    guide: 'gollum',
+                  },
+                ],
+                getHobbit: {
+                  name: 'Frodo',
+                },
+                getTomBombadil: {
+                  times: 2,
+                },
+              }),
+            ),
+        );
+      });
+
+      describe('one of the paramter groups failes parameter validation', () => {
+        it('does not execute a request for that parameter group', async () => {
+          mockValidateParameters.mockImplementation(
+            (operationId, parameters) =>
+              new Promise((resolve) => {
+                if (
+                  operationId === 'walkIntoMordor' &&
+                  Object.keys(parameters)[0] === 'guide'
+                )
+                  throw new TypeMismatchError(
+                    ['parameters', 'guide', 'example'],
+                    'string',
+                    'number',
+                  );
+
+                resolve();
+              }),
+          );
+          await expect(async () => {
+            await Positive.run(['http://urldoesnotmatter.com']);
+          }).rejects.toThrow('1 operation failed');
+
+          expect(mockExecute).not.toHaveBeenCalledWith('walkIntoMordor', {
```

Comment:
i thought about that. It seemed unnecessary but it would catch a bug where _no_ executes are called if even 1 fails. I don't mind adding it in if you think we should cover it.